### PR TITLE
Manage BrokerHost in NodeController.

### PR DESCRIPTION
**Pull request by hyowon**
When launching a child process in Castanets mode, we assigns a
unique negative process handle. This CL is to manage BrokerHosts
by the map with the handle as the key value in NodeController.

### CODE REVIEW

File: mojo/core/node_controller.cc [hunk 1]
```diff
@@ -322,6 +322,13 @@ bool NodeController::SyncSharedBuffer(
     size_t sync_size) {
   if (broker_)
     return broker_->SyncSharedBuffer(guid, offset, sync_size);
+  else if (!broker_hosts_.empty()) {
+    // TODO(hw1008.kim): Assume there is only one connected node. In multiple
+    // nodes scenario, we have to find a proper broker host for the
+    // corresponding guid.
+    broker_hosts_.begin()->second->SyncSharedBuffer(guid, offset, sync_size);
+    return true;
+  }
```
Comment (is46-kim) [inline mojo/core/node_controller.cc:333]: need a lock to access |broker_hosts_|?

File: mojo/core/node_controller.cc [hunk 2]
```diff
@@ -332,6 +339,13 @@ bool NodeController::SyncSharedBuffer(
     size_t sync_size) {
   if (broker_)
     return broker_->SyncSharedBuffer(mapping, offset, sync_size);
+  else if (!broker_hosts_.empty()) {
+    // TODO(hw1008.kim): Assume there is only one connected node. In multiple
+    // nodes scenario, we have to find a proper broker host for the
+    // corresponding guid.
+    broker_hosts_.begin()->second->SyncSharedBuffer(mapping, offset, sync_size);
+    return true;
+  }
```
Comment (is46-kim) [inline mojo/core/node_controller.cc:352]: ditto

File: mojo/core/node_controller.cc (hunk 1 above)
Comment (hyowon) [reply mojo/core/node_controller.cc:333]: RIGHT! THANKS!

File: mojo/core/node_controller.cc (hunk 2 above)
Comment (hyowon) [reply mojo/core/node_controller.cc:352]: done

**Review by is46-kim — APPROVED** [review]

**Review by Sunwoo-Nam — APPROVED** [review]
LGTM!!!!!!

**Pull request merged**